mesonbuild/backend/ninjabackend.py

```diff
@@ -2355,7 +2355,12 @@ def generate_static_link_rules(self):
                 # ranlib, not to ar
                 cmdlist.extend(args)
                 args = []
-                cmdlist.extend(['&&', 'ranlib', '-c', '$out'])
+                # Ensure that we use the user-specified ranlib if any, and
+                # fallback to just picking up some ranlib otherwise
+                ranlib = self.environment.lookup_binary_entry('ranlib')
```

Comment:
Doesn't this need the for_machine argument?

Reply:
Ran into an error.

```
    ranlib = self.environment.lookup_binary_entry('ranlib')
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
TypeError: Environment.lookup_binary_entry() missing 1 required positional argument: 'name'
```

Reply:
Oops, I refactored and forgot that.

Reply:
@vanc please try again